src/ConfigSpace/hyperparameters/hyperparameter.py

```diff
@@ -0,0 +1,780 @@
+                f"\n{self}",
+            )
+
+        return self._neighborhood(f64(vector), n, std=std, seed=seed)
+
+    def get_max_density(self) -> float:
+        """Get the maximum density of the hyperparameter distribution."""
+        return self._vector_dist.max_density()
+
+    def neighbors_values(
+        self,
+        value: ValueT | DType,
+        n: int,
+        *,
+        std: float | None = None,
+        seed: np.random.RandomState | None = None,
+    ) -> Array[DType]:
+        """Sample neighbors of a value.
+
+        Args:
+            value:
+                The value to sample neighbors from.
+            n:
+                The number of **unique** neighbors to sample.
+
+                !!! warning
+
+                    If there are less than `n` legal neighbors, then all legal
+                    neighbors are returned, which is some number less than `n`.
+            std:
+                The standard deviation of the neighborhood. If `None` the
+                neighborhood is deterministic. Defaults to `None`.
+
+                !!! warning
+
+                    Hyperparameter subclasses are under no obligation to use
+                    this if it does not make sense, i.e. for an
+                    [`OrdinalHyperparameter`][ConfigSpace.hyperparameters.OrdinalHyperparameter]
+                    or a
+                    [`CategoricalHyperparameter`][ConfigSpace.hyperparameters.CategoricalHyperparameter].
+
+            seed:
+                The random state to use for sampling. If `None` the global
+                random state is used. Defaults to `None`.
+
+        Returns:
+            The sampled neighbors in value space.
+        """
+        vector = self.to_vector(value)
+        return self.to_value(
+            vector=self.neighbors_vectorized(vector, n, std=std, seed=seed),
+        )
+
+    def pdf_vector(self, vector: Array[f64]) -> Array[f64]:
+        """Get the probability density of an array vectorized values.
+
+        Args:
+            vector:
+                The vectorized values to get the probability density of.
+
+        Returns:
+            The probability density of the vectorized values. Where vectorized
+            values are not legal, the probability density is zero.
+        """
+        legal_mask: Array[f64] = self.legal_vector(vector).astype(f64)
+        return self._vector_dist.pdf_vector(vector) * legal_mask
+
+    def pdf_values(
+        self,
+        values: Sequence[ValueT | DType] | Array[DType],
+    ) -> Array[f64]:
+        """Get the probability density of an array of values.
+
+        Args:
+            values:
+                The values to get the probability density of.
+
+        Returns:
+            The probability density of the values. Where values are not legal,
+            the probability density is zero.
+        """
+        # TODO(eddiebergman): Backwards compatible restriction, why this restriction?
```

Comment:
This would be great to know ;)

Reply:
Will leave it in now unless it's a problem at some point